### ISSUE TITLE
Socket.io doesn't upgrade to websocket

### ISSUE THREAD

**Issue by orozkoinsider** [socketio/socket.io-client-swift#421]
Hi to the socket.io community, 

I am trying to connect my app to a socket.io server in aws, behind an elb and a haproxy. The configuration with the sockets should be ok because the web and android clients are already upgrading to websockets. 

The connection fallbacks to polling all the time, and I get multiple connect events and sometimes I still get Session ID unknown errors. This happens when the server sends a 500 error and the app tries to reconnect, receiving error 400. 

The strange thing is that as soon as I launch the app, I get an endEncountered event in the public func stream(_ aStream: Stream, handle eventCode: Stream.Event) function in the WebSocket.swift file, even if I don´t force the websockets. 

<img width="1331" alt="captura de pantalla 2016-07-13 a las 10 48 58" src="https://cloud.githubusercontent.com/assets/645151/16797789/5ec6d54a-48e9-11e6-9a8c-fc8409cedc8e.png">

The issue is happening with the 6.1.4 and the swift 3 versions of the socket.io-client-swift. The version in the servers socket.io should be the 1.4.8 or 1.4.7.

Can I contact, call or skype to anybody and try to debug this behavior? 

Thanks

**Comment by nuclearace** [comment]
This is because the websocket library does not yet support tunneling through proxies. There's an open pull request that adds support for it, but until then it will continue to fail.

https://github.com/daltoniam/Starscream/pull/202

**Comment by orozkoinsider** [comment]
Thanks @nuclearace I just tried @oldshuren's socket.io-client-swift fork with proxy connect and almost works. At least I can see the wss connection even if the server is closing it. 

<img width="1287" alt="captura de pantalla 2016-07-13 a las 16 01 20" src="https://cloud.githubusercontent.com/assets/645151/16806066/3e5c056c-4913-11e6-8ac7-314d42d66d4a.png">

**Comment by nuclearace** [comment]
Hm, I haven't (and probably won't) had time to set up a test setup. But once that pull request is merged and I update the dependency I'll look into a little bit more.

**Comment by orozkoinsider** [comment]
I also will try to make it work, thanks!

**Comment by Dwar3xwar** [comment]
Any updates on this issue?

**Comment by QaiserAbbas** [comment]
hi, any update or pull is merged or not ? Its long time now on this Issue

**Comment by jlgl10nes** [comment]
Any updates on this issue? I see it "on hold"
For a scalable use of socket.io it's almost a mandatory enhancement to use a load balancer (proxy), the alternatives are painful.

**Comment by nuclearace** [comment]
It's still blocked by the dependency.

**Comment by megamingus** [comment]
Any updates?

**Comment by slumtrimpet** [comment]
So um... real talk... is this actually still an issue in 2000 freaking 23?

**Comment by ivanornes** [comment]
Bad proxy, no issue in the library
